tests/Unit/Evolution/Systems/ScalarTensor/Test_TimeDerivativeTerms.cpp

```diff
@@ -0,0 +1,259 @@
+
+#include "Framework/TestingFramework.hpp"
+
+#include <cstddef>
+#include <memory>
+#include <random>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Evolution/Systems/CurvedScalarWave/System.hpp"
+#include "Evolution/Systems/CurvedScalarWave/Tags.hpp"
+#include "Evolution/Systems/CurvedScalarWave/TimeDerivative.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/DampedHarmonic.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/Tags/GaugeCondition.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"
+#include "Evolution/Systems/ScalarTensor/System.hpp"
+#include "Evolution/Systems/ScalarTensor/TimeDerivativeTerms.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "Helpers/DataStructures/MakeWithRandomValues.hpp"
+#include "PointwiseFunctions/GeneralRelativity/GeneralizedHarmonic/DerivSpatialMetric.hpp"
+#include "PointwiseFunctions/GeneralRelativity/GeneralizedHarmonic/ExtrinsicCurvature.hpp"
+#include "PointwiseFunctions/GeneralRelativity/GeneralizedHarmonic/SpatialDerivOfLapse.hpp"
+#include "PointwiseFunctions/GeneralRelativity/GeneralizedHarmonic/SpatialDerivOfShift.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TypeTraits/IsA.hpp"
+
+namespace {
+template <typename ComputeVolumeTimeDerivativeTerms, size_t Dim,
+          typename EvolvedTagList, typename FluxTagList, typename TempTagList,
+          typename GradientTagList, typename ArgTagList>
+struct ComputeVolumeTimeDerivativeTermsHelper;
+
+template <typename ComputeVolumeTimeDerivativeTerms, size_t Dim,
+          typename... EvolvedTags, typename... FluxTags, typename... TempTags,
+          typename... GradientTags, typename... ArgTags>
+struct ComputeVolumeTimeDerivativeTermsHelper<
+    ComputeVolumeTimeDerivativeTerms, Dim, tmpl::list<EvolvedTags...>,
+    tmpl::list<FluxTags...>, tmpl::list<TempTags...>,
+    tmpl::list<GradientTags...>, tmpl::list<ArgTags...>> {
+  template <typename EvolvedVariables, typename FluxVariables,
+            typename TemporaryVariables, typename GradientVariables,
+            typename ArgumentVariables>
+  static void apply(
+      const gsl::not_null<EvolvedVariables*> dt_vars_ptr,
+      [[maybe_unused]] const gsl::not_null<FluxVariables*> volume_fluxes,
+      const gsl::not_null<TemporaryVariables*> temporaries,
+      const GradientVariables& partial_derivs,
+      const ArgumentVariables& time_derivative_args) {
+    ComputeVolumeTimeDerivativeTerms::apply(
+        make_not_null(&get<::Tags::dt<EvolvedTags>>(*dt_vars_ptr))...,
+        make_not_null(&get<FluxTags>(*volume_fluxes))...,
+        make_not_null(&get<TempTags>(*temporaries))...,
+        get<GradientTags>(partial_derivs)..., [](const auto& t) -> const auto& {
+          if constexpr (tt::is_a_v<std::unique_ptr,
```

Comment:
do the systems have any arg tags that are unique pointers?

Reply:
Apparently `gh::gauges::GaugeCondition` has a unique pointer